File: project/src/main/java/com/moyeoba/project/security/jwt/PermittedPath.java
package com.moyeoba.project.security.jwt;

public class PermittedPath {
    private final static String[] OPEN_PATH = new String[] {
            "/test/*", "/login/*", "/swagger-ui/*", "/configuration/ui", "/v3/api-docs/*"
    };

    public static boolean isOpen(String url) {
        for(String path: OPEN_PATH) {
            boolean result = true;

            String openUrl = "";
            if(path.split("/\\*").length != 0) {
                openUrl = path.split("/\\*")[0];
            } else {
                openUrl = path;
            }



            for(int i =0; i < openUrl.length(); i++) {
                if(openUrl.charAt(i) != url.charAt(i)) {
                    result = false;
                    break;
                }
            }

            if(result) return true;
        }

        return false;
    }
}

File: src/main/kotlin/com/moyeoba/moyeoba/jwt/PermittedPath.kt
package com.moyeoba.moyeoba.jwt

object PermittedPath {
    private val OPEN_PATH = arrayOf("/test/*", "/user/*")

    fun isOpen(url: String): Boolean {
        for (path in OPEN_PATH) {
            var result = true

            val openUrl = if(path.split("/*").isNotEmpty()) {
                //println("split: ${path.split("/*")[0]}")
                path.split("/*")[0]

            } else {
                path
            }

            for (i in openUrl.indices) {
                if (openUrl[i] != url[i]) {
                    //println("${openUrl[i]} is different from ${url[i]}")
                    result = false
                    break
                }
            }

            if (result) return true
        }

        return false
    }
}